src/pages/AccountRequest/RequestForm.tsx

```diff
@@ -0,0 +1,122 @@
+import FormInput from "components/Form/FormInput";
+import FormSelect from "components/Form/FormSelect";
+import { Form, Formik, FormikHelpers } from "formik";
+import React from "react";
+import { IAccountRequestForm } from "./accountUtil";
+import * as Yup from "yup";
+import useAPI from "hooks/useAPI";
+import { Button, Col, InputGroup } from "react-bootstrap";
+import { HttpMethod } from "utils/httpMethods";
+import { useLoaderData } from "react-router-dom";
+const initialValues: IAccountRequestForm = {
+  role_id: 0,
+  username: "",
+  full_name: "",
+  email: "",
+  institution_id: 0,
+  introduction: "",
+};
+
+const validationSchema = Yup.object({
+  role_id: Yup.number().required("Required"),
+  username: Yup.string()
+    .required("Required")
+    .lowercase("Username must be lowercase")
+    .min(3, "Username must be at least 3 characters")
+    .max(20, "Username must be at most 20 characters"),
+  full_name: Yup.string().required("Required"),
+  email: Yup.string().required("Required").email("Invalid email format"),
+  institution_id: Yup.number().required("Required"),
+  introduction: Yup.string().required("Required"),
+});
+
+const RequestForm: React.FC = () => {
+  const { roles, institutions }: any = useLoaderData();
+  const { sendRequest } = useAPI();
+
+  const onSubmit = async (
+    values: IAccountRequestForm,
+    submitProps: FormikHelpers<IAccountRequestForm>
+  ) => {
+    let method: HttpMethod = HttpMethod.POST;
+    let url: string = "/account_requests";
+    sendRequest({
```

Comment:
You need to handle errors as well, so what will happen if something went wrong? You should tell users what's wrong.

```typescript
  const { error,  sendRequest } = useAPI();
  const dispatch = useDispatch();

// Error alert
  useEffect(() => {
    if (error) {
      dispatch(alertActions.showAlert({ variant: "danger", message: error }));
    }
  }, [error, dispatch]);
```